contents/javascript/Javascript-framework-Angular.md

```diff
@@ -30,12 +30,22 @@ Consider a simple web application that displays the name of a person as shown be
     The sandbox above is editable. You can enter your name in the text box and see how the UI is automatically updated. You can also view and edit the code by clicking on the Both button. 
 </box>
 
-The UI of an Angular app is defined in _templates_ which contain a mix of HTML code and HTML-like custom tags.
+The UI of an Angular app is defined in _templates_ which contain a mix of HTML code and HTML-like custom tags. The template for the above example is given below.
+
+``` html
+<h1>About Me!</h1>
+<h2>Hello World! My name is {hero.name}</h2>
+<div>
+<label>NAME:
+  <input [(ngModel)]="hero.name"/>
+</label>
+</div>
+```
```

Comment:
Down below it is mentioned that `selector: app-heroes` connects the component to this template. Did I understand that correctly? If yes, this code has no mention of `app-heroes`. How come? If I misunderstood it, that sentence might need rephrasing to make the meaning clearer.

Reply:
So if you look in the `app.component.html` file of this project you will notice the following template:
```html
<app-heroes></app-heroes>
```
Seeing this <app-heros> tag, Angular knows that it has to render the `HeroComponent` template which is the template given as an example.

The idea is, whenever this `<app-heros>` tag is used in any template across an Angular project, the `heroComponent` will be rendered in the UI.

Reply:
Well, that isn't mentioned in the article at all. The reader will not be able to make the connection.

Reply:
I'll rephrase this

Reply:
>The use of selector: `app-person` is how Angular identifies this component in a template. For instance, if the `<app-person>` tag is encountered in any template across the project, Angular instantiates the `PersonComponent` and renders the view defined by the template `./person.component.html`

Does this explain it well enough?

Reply:
It appears the example involves three key files but only two were explained. The file containing the `<app-person>` tag was not explained. Hence the confusion. Also, when you give code examples, perhaps state the file name as well?

Reply:
Added the filenames 👍